File: DSAlgorithms/Leetcode/c++/2_pointers/680-valid_palindrome2.cpp
class Solution {
public:
    bool validPalindrome(string s) {
       int left = 0;
       int right = s.size() - 1;
       while(left < right) 
       {
           if(s[left] != s[right])
            {
                std::string skip_left = s.substr(left + 1, right - left); //substring without left char
                std::string skip_right = s.substr(left, right - left); //substring without right char   
                std::string reversed_left = skip_left;
                std::string reversed_right = skip_right;
                reverse(reversed_left.begin(), reversed_left.end());
                reverse(reversed_right.begin(), reversed_right.end());
                return (skip_left == reversed_left || skip_right == reversed_right);
            }
            left ++;
            right --;
       }
       return true;
    }
};

File: DSAlgorithms/Leetcode/ts/src/2_pointers/680-valid_palindrome2.ts
const validPalindrome = (s: string): boolean =>{
    let left = 0;
    let right = s.length - 1;
    while(left < right)
    {
        if(s[left] !== s[right])
        {
            let skip_left: string = s.substring(left + 1, right + 1);
            let skip_right: string = s.substring(left, right);
            let left_reversed: string = skip_left;
            let right_reversed: string = skip_right;
            left_reversed = left_reversed.split("").reverse().join("");
            right_reversed = right_reversed.split("").reverse().join("");
            return (skip_left === left_reversed || skip_right === right_reversed);
        }
        left ++;
        right --;
    }
    return true;
    };